src/js/models/documentexports.js

```diff
@@ -48,6 +49,35 @@ define(function (require, exports, module) {
         layerExportsMap: Immutable.Map()
     });
 
+    Object.defineProperties(DocumentExports.prototype, objUtil.cachedGetSpecs({
+        /**
+         * Produce an array of ExportAssets for the root of this document, converted to a JS object
+         * This is used for serializing the data to Ps metadata store
+         *
+         * @return {Array.<object>} An array of ExportAsset-like JS objects
+         */
+        rootExportsArray: function () {
+            if (this.rootExports && this.rootExports.size > 0) {
+                return this.rootExports.toJS();
+            } else {
+                return [];
+            }
+        },
+
+        /**
+         * Get a list of layer IDs that have a non-empty set of configured export assets
+         *
+         * @return {Immutable.IndexedSeq.<number>}
+         */
+        layerIDsWithExports: function () {
```

Comment:
I'm curious about why you chose a Seq here. I usually think of Seqs as useful when incrementally constructing complicated data structures because they obviate construction of the intermediate data structures. But it doesn't seem like that's how you're using this?